src/server/worker/master.go

```diff
@@ -472,6 +478,19 @@ func (a *APIServer) waitJob(pachClient *client.APIClient, jobInfo *pps.JobInfo,
 			return nil // retry again
 		})
 	}()
+	// Get the stats commit.
+	var statsCommit *pfs.Commit
+	if jobInfo.EnableStats {
+		ci, err := pachClient.InspectCommit(jobInfo.OutputCommit.Repo.Name, jobInfo.OutputCommit.ID)
+		if err != nil {
+			return err
+		}
+		for _, commitRange := range ci.Subvenance {
```

Comment:
I was just reading this in `master` and I'm a little baffled by this - the commit has its own repo as subvenance for stats?  Also, we check the repo on both the lower and upper commit range - can these ever be different?  I realize this was existing code and you just moved it, but I'm just curious.

Reply:
I am actually not too sure I can answer these questions confidently (but @jdoliner or @msteffen might). For your first question, I think we operate on branch provenance/subvenance so having a commit with the same repo in the subvenance is fine and fits the model. As far as the lower/upper stuff is concerned, I could use an explanation as well.

Reply:
Stats: yes, the model is that branches are provenant on other branches, and the stats branch is provenant on the output branch (so that whenever a new commit is created in the output branch, a new commit is created in the stats branch as well): https://github.com/pachyderm/pachyderm/blob/845caa20fcc8a24067d1e4461ff09932a9a29a6c/src/server/pps/server/api_server.go#L2080

Lower/Upper: No, there's not really any way for Lower/Upper to be in different repos. The only way for a subvenance range to change is if a new subvenant commit has Upper as its parent: https://github.com/pachyderm/pachyderm/blob/master/src/server/pfs/server/driver.go#L3907

Reply:
:+1: this is in keeping with my understanding as well.